framework/testutil/address/address.go

```diff
@@ -14,10 +14,10 @@ type PeerAddresser interface {
 
 // BuildInternalPeerAddressList constructs a comma-separated list of internal peer addresses from the given nodes.
 // It returns an error if any node fails to provide a peer address.
-func BuildInternalPeerAddressList(ctx context.Context, peer []PeerAddresser) (string, error) {
-	addrs := make([]string, 0, len(peer))
+func BuildInternalPeerAddressList[T PeerAddresser](ctx context.Context, peers []T) (string, error) {
```

Comment:
small improvement that allows for cleaner usage without needing to create a new slice of PeerAddresser each time.